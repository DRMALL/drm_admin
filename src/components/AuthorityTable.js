

import React from 'react'
import Table from 'antd/lib/table'
import { browserHistory } from 'react-router'
import TableTime from './TableTime'
import moment from 'moment'
import dispatch from '../actions/dispatch'
import store from '../commons/store'
import TableFilter from './TableFilter'
import Modal from 'antd/lib/modal'
import authorityDelById from '../actions/authorityDelById'

export default props => {
  const { authorityArray, userNameArr, machineNameArr, authModal } = props
  const columns = [{
    title: '用户',
    dataIndex: 'user.name',
    width: '20%',
    filterDropdown: ( <TableFilter changeFun={ handleChangexx } arr={ userNameArr } /> )
  },{
    title: '查看权限',
    dataIndex: 'canView',
    width: '10%',
    render: text =>
    <div>{ text ? '有' : '无' }</div>
  },{
    title: '监控权限',
    dataIndex: 'canMonitor',
    width: '10%',
    render: text =>
    <div>{ text ? '有' : '无' }</div>
  },{
    title: '设备名称',
    dataIndex: 'device.name',
    width: '25%',
    filterDropdown: ( <TableFilter changeFun={ handleChangeMachine } arr={ machineNameArr } /> )
  },{
    title: '修改时间',
    dataIndex: 'updatedAt',
    width: '10%',
    render: text => TableTime(moment(text).format('YYYY-MM-DD HH:mm'), 15)
  },{
    title: '操作',
    key:'action',
    width: '10%',
    render: (text, record) => (
      <div className='table-icon-content' >
        <div className='table-icon-item'  style={{ color: '#579df2' }}
             onClick={ navToEditAuth } data-auth-id={ record._id } >
          <i className="iconfont icon-compile trouble-table-icon"  style={{ color: '#0068d2', fontSize:25 }} ></i>
        </div>
        <div className='table-icon-item' onClick={ delAuth } data-auth-id={ record._id }  >
          <i className="iconfont icon-shanchu del-icon" style={{ fontSize: 20 }} ></i>
        </div>
      </div>
      )
  }]
  return(
    <div>
      <Table columns={columns} dataSource={authorityArray} rowKey='_id' />
      <Modal title='提示' visible={ authModal } okText='确定' cancelText='取消'
             onOk={ authorityDelById } onCancel={ cancleDelAuth } >
             <p>确定删除此权限？</p>
      </Modal>
    </div>
    )
}

function navToEditAuth(e){
  const { authId } = e.currentTarget.dataset
  localStorage.setItem('authId', authId)
  browserHistory.push('/authority/edit')
}

function handleChangexx(e){
  const { TauthorityArray } = store.getState().authority
  if(e){
  let arr = TauthorityArray.filter(item => item.user._id === e )
  dispatch('AUTHORITY_FILTER_VALUE_TABLE', arr)
  }
  else dispatch('AUTHORITY_FILTER_VALUE_TABLE', TauthorityArray)
}

function handleChangeMachine(e){
  const { TauthorityArray } = store.getState().authority
  if(e){
  let arr = TauthorityArray.filter(item => item.device._id === e )
  dispatch('AUTHORITY_FILTER_VALUE_TABLE', arr)
  }
  else dispatch('AUTHORITY_FILTER_VALUE_TABLE', TauthorityArray)
}

function delAuth(e){
  const { authId } = e.currentTarget.dataset
  dispatch( 'AUTHORITY_DEL_GET_AUTHID', authId )
}

function cancleDelAuth(){
  dispatch('AUTHORITY_CANCLE_DEL_AUTH')
}
