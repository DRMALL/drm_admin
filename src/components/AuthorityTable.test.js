

import AuthorityTable from './AuthorityTable'
// import renderer from 'react-test-renderer'
import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'


it('AuthorityTable test', () => {
  const authorityArray = [{
canMonitor:true,
canView:true,
createdAt:"2017-07-28T11:46:32.667Z",
device:{ name: "设备名称", number: 'a42355', _id: 'dgtre2533' },
updatedAt: "2017-08-02T02:07:54.294Z",
user: { name: "李四", _id: 'asffdg23235' },
__v:0,
_id: 'frghs3453464',
},{
canMonitor:true,
canView:false,
createdAt:"2017-07-28T11:46:32.667Z",
device:{ name: "设备名称2", number: 'a42356', _id: 'dgtre2553' },
updatedAt: "2017-08-02T02:07:54.294Z",
user: { name: "张三", _id: 'asffdg232242' },
__v:0,
_id: 'frghs3455367',
},{
canMonitor:false,
canView:true,
createdAt:"2017-07-28T11:46:32.667Z",
device:{ name: "设备名称3", number: 'a42357', _id: 'dgtre2524' },
updatedAt: "2017-08-02T02:07:54.294Z",
user: { name: "王二", _id: 'asffdg23224' },
__v:0,
_id: 'frghs3453473',
}]

const userNameArr = [
      {name: '张三', _id: 'asffdg232242'},
      { name: "王二", _id: 'asffdg23224' },
      { name: "李四", _id: 'asffdg23235' }
         ]

const machineNameArr = [
      { name: "设备名称", number: 'a42355', _id: 'dgtre2533' },
      { name: "设备名称2", number: 'a42356', _id: 'dgtre2553' },
      { name: "设备名称3", number: 'a42357', _id: 'dgtre2524' }]

const wrapper = render(<AuthorityTable authorityArray={ authorityArray }
                          userNameArr={ userNameArr }
                          machineNameArr={ machineNameArr } />)
        expect(toJson(wrapper)).toMatchSnapshot()
})
