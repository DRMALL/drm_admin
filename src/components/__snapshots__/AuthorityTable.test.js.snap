// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorityTable test 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-large ant-table-scroll-position-left"
        >
          <div
            class="ant-table-content"
          >
            <div
              class="ant-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col
                    style="width:20%;min-width:20%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:25%;min-width:25%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class=""
                    >
                      <span>
                        用户
                        <i
                          class="anticon anticon-filter ant-dropdown-trigger"
                          title="筛选"
                        />
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        查看权限
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        监控权限
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        设备名称
                        <i
                          class="anticon anticon-filter ant-dropdown-trigger"
                          title="筛选"
                        />
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        修改时间
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        操作
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      李四
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        有
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        有
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      设备名称
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        <div>
                          2017-08-02
                        </div>
                        <div
                          style="margin-left:15px;"
                        >
                          10:07
                        </div>
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div
                        class="table-icon-content"
                      >
                        <div
                          class="table-icon-item"
                          data-auth-id="frghs3453464"
                          style="color:#579df2;"
                        >
                          <i
                            class="iconfont icon-compile trouble-table-icon"
                            style="color:#0068d2;font-size:25px;"
                          />
                        </div>
                        <div
                          class="table-icon-item"
                          data-auth-id="frghs3453464"
                        >
                          <img
                            alt="删除"
                            src="dels.png"
                            style="width:20px;height:20px;"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      张三
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        无
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        有
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      设备名称2
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        <div>
                          2017-08-02
                        </div>
                        <div
                          style="margin-left:15px;"
                        >
                          10:07
                        </div>
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div
                        class="table-icon-content"
                      >
                        <div
                          class="table-icon-item"
                          data-auth-id="frghs3455367"
                          style="color:#579df2;"
                        >
                          <i
                            class="iconfont icon-compile trouble-table-icon"
                            style="color:#0068d2;font-size:25px;"
                          />
                        </div>
                        <div
                          class="table-icon-item"
                          data-auth-id="frghs3455367"
                        >
                          <img
                            alt="删除"
                            src="dels.png"
                            style="width:20px;height:20px;"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      王二
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        有
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        无
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      设备名称3
                    </td>
                    <td
                      class=""
                    >
                      <div>
                        <div>
                          2017-08-02
                        </div>
                        <div
                          style="margin-left:15px;"
                        >
                          10:07
                        </div>
                      </div>
                    </td>
                    <td
                      class=""
                    >
                      <div
                        class="table-icon-content"
                      >
                        <div
                          class="table-icon-item"
                          data-auth-id="frghs3453473"
                          style="color:#579df2;"
                        >
                          <i
                            class="iconfont icon-compile trouble-table-icon"
                            style="color:#0068d2;font-size:25px;"
                          />
                        </div>
                        <div
                          class="table-icon-item"
                          data-auth-id="frghs3453473"
                        >
                          <img
                            alt="删除"
                            src="dels.png"
                            style="width:20px;height:20px;"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-prev"
            tabindex="0"
            title="上一页"
          >
            <a />
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-next"
            tabindex="0"
            title="下一页"
          >
            <a />
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
